src/notation/internal/notationparts.cpp

```diff
@@ -1217,6 +1217,10 @@ void NotationParts::appendStaves(Part* part, const mu::instruments::Instrument&
 
         insertStaff(staff, staffIndex);
     }
+
+    int firstStaffIndex = part->staff(0)->idx();
```

Comment:
Probably you need to add a check for emptiness here:

if (part->staves().isEmpty()) {
return;
}
int firstStaffIndex = ...
...

Reply:
This will only happen if the `Instrument` has no staves which is wrong in the first place (instruments have at least one staff).
However since methods in `NotationParts` are a kind of "high level" methods, a tests might do no harm here.
Done.